packages/mobile/src/escrow/EscrowedPaymentListItem.tsx

```diff
@@ -73,42 +86,36 @@ export class EscrowedPaymentListItem extends React.PureComponent<Props> {
   getCTA = () => {
     const { t } = this.props
     const ctas = []
-    if (this.getDisplayName()) {
-      ctas.push({
-        text: t('global:remind'),
-        onPress: this.onRemind,
-      })
-    }
+    ctas.push({
+      text: t('global:remind'),
+      onPress: this.onRemind,
+    })
     ctas.push({
       text: t('global:reclaim'),
       onPress: this.onReclaimPayment,
     })
     return ctas
   }
 
-  getDisplayName() {
-    const { payment } = this.props
-    // TODO(Rossy) Get contact number from recipient cache here
-    return payment.recipientPhone
-  }
-
   render() {
-    const { t, payment } = this.props
-    const mobile = this.getDisplayName() || t('global:unknown').toLowerCase()
+    const { t, payment, phoneRecipientCache } = this.props
     const amount = {
       value: divideByWei(payment.amount),
       currencyCode: CURRENCIES[CURRENCY_ENUM.DOLLAR].code,
     }
+    const recipient: Recipient = phoneRecipientCache[payment.recipientPhone] ?? {
+      e164PhoneNumber: payment.recipientPhone,
+    }
 
     return (
       <View style={styles.container}>
         <RequestMessagingCard
-          title={t('escrowPaymentNotificationTitle', { mobile })}
+          title={t('escrowPaymentNotificationTitle', { mobile: payment.recipientPhone })}
           amount={<CurrencyDisplay amount={amount} />}
           details={payment.message}
           icon={
             <ContactCircle
-              name={mobile}
+              recipient={recipient}
               // TODO: Add thumbnailPath={}
```

Comment:
Since we're passing the recipient this TODO for `thumbnailPath` is already covered, right?